Prepare for release 0.3.0 (#29)
logicalclocks/maggy: README.rst

Maggy
=====

|Downloads| |PypiStatus| |PythonVersions| |Docs|

Maggy is a framework for efficient asynchronous optimization of expensive
black-box functions on top of Apache Spark. Compared to existing frameworks,
maggy is not bound to stage based optimization algorithms and therefore it is
able to make extensive use of early stopping in order to achieve efficient
resource utilization.

Right now, maggy supports asynchronous hyperparameter tuning of machine
learning and deep learning models, and ablation studies on neural network
layers as well as input features.

Moreover, it provides a developer API that allows advanced usage by
implementing custom optimization algorithms and early stopping criteria.

To accomodate asynchronous algorithms, support for communication between the

> Driver and Executors via RPCs was added. The Optimizer that guides

Reviewer: was added -> maggy supports communication between the Driver and Executors via RPCs.

hyperparameter search is located on the Driver and it assigns trials to
Executors. Executors periodically send back to the Driver the current

> performance of their trial, and the Optimizer can decide to early-stop its

Reviewer: ```suggestion
performance of their trial, and the Optimizer can decide to early-stop
```

> ongoing trial, followed by sending the Executor with a new trial.

Reviewer: ```suggestion
any ongoing trial and send the Executor a new trial instead.
```
